ClinGen/clincoded
# Connects to #827. Codon data display.

This PR consists of changes pertinent to the handling of UI display in the variant codon table on the computational tab.

**Technical details:**
1. Abstracted the repetitive code from index.js and created a helper method to get HGVS notation in the formats required by either myvariant.info or Ensembl VEP.
2. Created a helper method to set the primary transcript based on the logical filtering of properties such as 'canonical', 'source' and 'hgvsc' in the Ensembl VEP response.
3. Added '&canonical=1&merged=1' params to Ensembl VEP API request to expose **canonical** flags and **RefSeq** transcripts in response.
4. Decoupled the Eutils/Esearch combo fetch method since the Esearch method for codon data relies on Ensembl VEP response going forward.
5. Updated the handling of UI display in the variant codon table given different test cases.

**Steps to test:**
Via the variant selection interface, try the following example IDs. Verify the expected display given the different test cases.

Test case 1 - Variant does not exist in ClinVar.
Expected display - **ClinVar search for this variant is currently not available.**
Example IDs: CA501097, CA501096, CA501104

Test case 2 - Protein change data, such as amino acid and/or location, not found in ClinVar.
Expected display - **The current variant is in a non-coding region.**
Example IDs: 139214, CA003323

Test case 3 - Has protein change, but ClinVar response returns count of 0.
Expected display - **No variants have been found at this codon in ClinVar.**
Example IDs: 

Test case 4 - Has protein change, and ClinVar response returns count of 1.
Expected display - **The current variant is the only variant found at this codon in ClinVar.**
Example IDs: 55847, 35824

Test case 5 - Has protein change, and ClinVar response returns count of more than 1. 
Expected display - **Additional ClinVar variants found in the same codon: count (Search ClinVar for variants in this codon)**
Example IDs: 35816, 53423
